### PR TITLE
feat(button): add iconOnly option

### LINKED ISSUE #1593: Add copy button component
Screen Shot 2020-07-13 at 12 08 33 PM

### PR DESCRIPTION
**Pull request by Donisius**
Closes #1593

I included the tooltip classes within the button directive as hostbindings for the assistive text because using the tooltip directive would place the tooltip into a placeholder component or to the body of the document. This conflicts with the html structure in https://react.carbondesignsystem.com/?path=/story/button--icon-button, unless you put the placeholder component inside the button itself which obviously isn't an option.

### CODE REVIEW

File: src/button/button.directive.ts [hunk 1]
```diff
@@ -34,6 +34,19 @@ export class Button implements OnInit {
 	 * Specify the size of the button
 	 */
 	@Input() size: ButtonSize;
+	/**
+	 * If assistive text is used, this specifies the placement.
+	 * Possible placements are `top`, `bottom`, `left`, `right`.
+	 * If assistive text is not used, this can be left undefined.
+	 */
+	@Input() assistiveTextPlacement: string;
```
Comment (zvonimirfras) [inline src/button/button.directive.ts:42]:
```suggestion
	@Input() assistiveTextPlacement: "top" | "bottom" | "left" | "right";
```

File: src/button/button.directive.ts [hunk 2]
```diff
@@ -34,6 +34,19 @@ export class Button implements OnInit {
...
+	/**
+	 * If assistive text is used, this specifies the alignment.
+	 * Possible alignments are `center`, `start`, `end`.
+	 * If assistive text is not used, this can be left undefined.
+	 */
+	@Input() assistiveTextAlignment: string;
```
Comment (zvonimirfras) [inline src/button/button.directive.ts:48]:
```suggestion
	@Input() assistiveTextAlignment: "center" | "start" | "end";
```

**Review by zvonimirfras — CHANGES_REQUESTED** [review]

**Review by cal-smith — CHANGES_REQUESTED** [review]

File: src/button/button.directive.ts [hunk 3]
```diff
@@ -34,6 +34,19 @@ export class Button implements OnInit {
 	 * Specify the size of the button
 	 */
 	@Input() size: ButtonSize;
+	/**
+	 * If assistive text is used, this specifies the placement.
+	 * Possible placements are `top`, `bottom`, `left`, `right`.
+	 * If assistive text is not used, this can be left undefined.
+	 */
+	@Input() assistiveTextPlacement: "top" | "bottom" | "left" | "right";
```
Comment (cal-smith) [inline src/button/button.directive.ts:42]: Is there a sane default we can use for this and the alignment input? Perhaps `top` `center`?

**Review by cal-smith — CHANGES_REQUESTED** [review]
Otherwise looks good

File: src/button/button.directive.ts (hunk 3 above)
Comment (Donisius) [reply src/button/button.directive.ts:42]: If we have an initial value then the tooltip classes would be applied to buttons that don't have the assistive text. It wouldn't really do anything besides just being there though.

Comment (cal-smith) [reply src/button/button.directive.ts:42]: Could we make it conditional on `this.iconOnly === true`? That way users don't have to set the placement and alignment for simple cases

Comment (Donisius) [reply src/button/button.directive.ts:42]: That's a good idea!

**Review by zvonimirfras — APPROVED** [review]

**Review by cal-smith — APPROVED** [review]

**Pull request merged**